src/dependency-registry.ts

```diff
@@ -0,0 +1,354 @@
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (!dependency) {
+    return;
+  }
+  const id = generateDependencyAdditionalRepositoryId(
+    dependency.id,
+    event.params._additionalRepositoryIndex
+  );
+
+  const additionalRepository = DependencyAdditionalRepository.load(id);
+  if (!additionalRepository) {
+    return;
+  }
+
+  const lastAdditionalRepositoryIndex = dependency.additionalRepositoryCount.minus(
+    BigInt.fromI32(1)
+  );
+  const lastAdditionalRepository = DependencyAdditionalRepository.load(
+    generateDependencyAdditionalRepositoryId(
+      dependency.id,
+      lastAdditionalRepositoryIndex
+    )
+  );
+  if (!lastAdditionalRepository) {
+    return;
+  }
+
+  additionalRepository.repository = lastAdditionalRepository.repository;
+  additionalRepository.save();
+
+  store.remove("DependencyAdditionalRepository", lastAdditionalRepository.id);
+}
+
+export function handleDependencyScriptUpdated(
+  event: DependencyScriptUpdated
+): void {
+  const dependencyType = event.params._dependencyType.toString();
+  const dependency = Dependency.load(dependencyType);
+  if (!dependency) {
+    return;
+  }
+
+  const dependencyRegistryContract = DependencyRegistryV0.bind(event.address);
+  let prevScriptCount = dependency.scriptCount.toI32();
+  const scriptCount = dependencyRegistryContract
+    .getDependencyScriptCount(event.params._dependencyType)
+    .toI32();
+
+  // Remove ProjectScripts that no longer exist on chain
+  if (prevScriptCount > scriptCount) {
+    for (let i = scriptCount; i < prevScriptCount; i++) {
+      const dependencyScript = DependencyScript.load(
+        generateDependencyScriptId(
+          event.params._dependencyType.toString(),
+          BigInt.fromI32(i)
+        )
+      );
+      if (dependencyScript) {
+        store.remove("DependencyScript", dependencyScript.id);
+      }
+    }
+  }
+
+  let scripts: string[] = [];
+  for (let i = 0; i < scriptCount; i++) {
+    let script = dependencyRegistryContract.getDependencyScriptAtIndex(
+      event.params._dependencyType,
+      BigInt.fromI32(i)
+    );
+
+    let dependencyScriptIndex = BigInt.fromI32(i);
+    let dependencyScript = new DependencyScript(
+      generateDependencyScriptId(dependencyType, dependencyScriptIndex)
+    );
+    dependencyScript.index = dependencyScriptIndex;
+    dependencyScript.dependency = dependencyType;
+    dependencyScript.script = script;
+    dependencyScript.save();
+
+    if (script) {
+      scripts.push(script);
+    }
+  }
+
+  const script = scripts.join("");
+
+  dependency.script = script;
+  dependency.scriptCount = scriptCount;
+
+  dependency.save();
+}
+
+export function handleSupportedCoreContractAdded(
+  event: SupportedCoreContractAdded
+): void {
+  const coreContract = Contract.load(
+    event.params._coreContractAddress.toHexString()
+  );
+
+  if (!coreContract) {
+    return;
+  }
+
+  coreContract.dependencyRegistry = event.address;
```

Comment:
@ryley-o How do you feel about this logic and the logic for the related remove function? Basically, I'm setting the dependency registry on the core contract based entirely on if its supported on the dependency registry contract. The alternative would be to create a bridge entity so that a Contract entity could be supported by multiple DependencyRegistry entities but since we're not really using that field anywhere currently I figured it wasn't worth the complexity.

Reply:
@yoshiwarab - this feels like a fine pattern to me. It is all a little awkward because V3 core contract stores a dependency registry address and pre-V3 cores don't have a field for it, and we end up over-defined on the V3 core. We have to choose one source of truth here, and I agree that letting the dependency registry control that field in our subgraph seems most appropriate from an indexing perspective. The V3 core contract's dependency registry field is more of a pointer for someone looking at only on-chain data.

Based on the above, everything in these handlers LGTM - but you will need to go into `/src/mapping-v3-core.ts` and modify the `refreshContract` function to not assign `contractEntity.dependencyRegistry` to prevent overwriting the values assigned here!

Reply:
Ah good call, thank you for the feedback!